stdlib/public/runtime/ImageInspectionELF.cpp

```diff
@@ -68,8 +72,14 @@ static SectionInfo getSectionInfo(const char *imageName,
   SectionInfo sectionInfo = { 0, nullptr };
   void *handle = dlopen(imageName, RTLD_LAZY | RTLD_NOLOAD);
   if (!handle) {
-    fatalError(/* flags = */ 0, "dlopen() failed on `%s': %s", imageName,
-               dlerror());
+#ifdef __ANDROID__
+      llvm::StringRef imagePath = llvm::StringRef(imageName);
+      if (imagePath.startswith("/system/lib") || (imageName && !imagePath.endswith(".so"))) {
+          warning(/* flags = */ 0, "dlopen() failed on `%s': %s", imageName, dlerror());
```

Comment:
I don't think this warning is necessary, since failure in this case may be expected. It'll just noise up logs.

Reply:
I prefer to keep the warning so hopefully somebody later would figure out the reason we are trying to link to all these libraries and try to avoid it. But I can remove it, no problem.

Reply:
I removed the warning in the final commit

Reply:
Thanks. It sounds like we're trying to load these private libraries because they're transitive dependencies of public libraries, but Android's dlopen is special-cased to reject attempts to open them explicitly, which is gross but a situation we can't do much about.